pkgs/tools/system/lshw/default.nix

```diff
@@ -1,27 +1,33 @@
-{ stdenv, fetchurl
-, withGUI? false, gtk? null, pkgconfig? null, sqlite? null  # compile GUI
- }:
-stdenv.mkDerivation rec {
+{ stdenv, lib, fetchurl, fetchpatch
+, withGUI ? false, gtk2, pkgconfig, sqlite # compile GUI
```

Comment:
@rasendubi I've got a bit of an experience to report when `? null`s can backfire. While doing a lot of renaming (gtk -> gtk2 and the like mostly) eval and nox missed a lot of packages with non-existing definition under `? null`.
Also I've seen packages using `? null`, but not using assertions.

When `import`ing packages without `? null`s you seem to have to explicitly add `a = null`, but it may prevent the situation like the before mentioned one.

Regarding difference between platforms -- I'm not sure how this could happen (btw, if you remember any examples, please share them; I'll write too if I'd stumble upon one), but if there is a situation like this -- it needs to be fixed anyway, right?